dev/src/cli/cli_create.ts

```diff
@@ -0,0 +1,202 @@
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as enquirer from 'enquirer';
+import { execSync } from 'child_process';
+
+const dirname = process.cwd();
+
+const AGENT_TS_TEMPLATE = (model: string) => `
+import {FunctionTool, LlmAgent} from '@google/adk';
+import {z} from 'zod';
+
+/* Mock tool implementation */
+const getCurrentTime = new FunctionTool({
+  name: 'get_current_time',
+  description: 'Returns the current time in a specified city.',
+  parameters: z.object({
+    city: z.string().describe("The name of the city for which to retrieve the current time."),
+  }),
+  execute: ({city}) => {
+    return {status: 'success', report: \`The current time in \$\{ city \} is 10: 30 AM\`};
+  },
+});
+
+export const rootAgent = new LlmAgent({
+  name: 'hello_time_agent',
+  model: '${model}',
+  description: 'Tells the current time in a specified city.',
+  instruction: \`You are a helpful assistant that tells the current time in a city.
+                Use the 'getCurrentTime' tool for this purpose.\`,
+  tools: [getCurrentTime],
+});
+`.trim()
+
+interface AgentCreationOptions {
+  agentName: string;
+  model: string;
+  apiKey: string;
+  project: string;
+  region: string;
+}
+
+interface ModelPromptResult {
+  model: string;
+}
+
+interface BackendPromptResult {
+  backend: 'vertex' | 'genai';
+}
+
+interface ProjectPromptResult {
+  project: string;
+}
+
+interface RegionPromptResult {
+  region: string;
+}
+
+interface ApiKeyPromptResult {
+  apiKey: string;
+}
+
+async function chooseModel(): Promise<ModelPromptResult> {
+  return await enquirer.prompt([
+    {
+      type: 'select',
+      name: 'model',
+      message: 'Choose a model for the root agent',
+      choices: [
+        { name: 'gemini-2.5-flash', value: 'gemini-2.5-flash' },
+        { name: 'gemini-2.5-pro', value: 'gemini-2.5-pro' },
+      ],
+    },
+  ]) as ModelPromptResult;
+}
+
+async function chooseBackend(): Promise<BackendPromptResult> {
+  return await enquirer.prompt([
+    {
+      type: 'select',
+      name: 'backend',
+      message: 'Choose a backend',
+      choices: [
+        { name: 'vertex', value: 'vertex' },
+        { name: 'googleai', value: 'googleai' },
+      ],
+    },
+  ]) as BackendPromptResult;
+}
+
+async function promptForProject(): Promise<ProjectPromptResult> {
+  return await enquirer.prompt([
+    {
+      type: 'input',
+      name: 'project',
+      message: 'Enter the Google Cloud Project ID',
+    },
+  ]) as ProjectPromptResult;
+}
+
+async function promptForRegion(): Promise<RegionPromptResult> {
+  return await enquirer.prompt([
+    {
+      type: 'input',
+      name: 'region',
+      message: 'Enter the Google Cloud Region',
+    },
+  ]) as RegionPromptResult;
+}
+
+async function promptForApiKey(): Promise<ApiKeyPromptResult> {
+  return await enquirer.prompt([
+    {
+      type: 'input',
+      name: 'apiKey',
+      message: 'Enter the Google AI API Key',
+    },
+  ]) as ApiKeyPromptResult;
+}
```

Comment:
Can you please inline that in the function to make the code smaller and simpler?